### PR TITLE
Update pylint to 2.8.2

### PR DESCRIPTION
**Pull request by schmitts**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-astroid/package.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ class PyAstroid(PythonPackage):
     homepage = "https://github.com/PyCQA/astroid"
     url      = "https://github.com/PyCQA/astroid/archive/astroid-1.4.5.tar.gz"
 
+    version('2.5.6', sha256='7fa655511c3726ce7058e9908065c385bcada47cb4cb31d5d6a7248b1fea58fa')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-astroid/package.py:16]: Dependencies need to be updated, it looks like they are now documented in `setup.cfg`

File: var/spack/repos/builtin/packages/py-pylint/package.py [hunk 2]
```diff
@@ -21,13 +22,16 @@ class PyPylint(PythonPackage):
...
+    depends_on('python@3.4:', when='@2:2.7.999', type=('build', 'run'))
+    depends_on('python@3.6:', when='@2.8.2:', type=('build', 'run'))
     depends_on('py-astroid', type=('build', 'run'))
     # note there is no working version of astroid for this
     depends_on('py-astroid@1.5.1:', type=('build', 'run'), when='@1.7:')
     depends_on('py-astroid@1.6:1.9', type=('build', 'run'), when='@1.9.4')
     depends_on('py-astroid@2.0:', type=('build', 'run'), when='@2.2.0:')
-    depends_on('py-astroid@2.2.0:2.999.999', type=('build', 'run'), when='@2.3.0:')
+    depends_on('py-astroid@2.2.0:2.999.999', type=('build', 'run'), when='@2.3.0:2.7.999')
+    depends_on('py-astroid@2.5.6:2.999.999', type=('build', 'run'), when='@2.8.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylint/package.py]:
```suggestion
    depends_on('py-astroid@2.5.6:2.6.999', type=('build', 'run'), when='@2.8.0:')
```

File: var/spack/repos/builtin/packages/py-pylint/package.py [hunk 3]
```diff
@@ -21,13 +22,16 @@ class PyPylint(PythonPackage):
...
+
     depends_on('py-six', type=('build', 'run'), when='@1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylint/package.py]: Six is no longer needed

File: var/spack/repos/builtin/packages/py-pylint/package.py [hunk 4]
```diff
@@ -21,13 +22,16 @@ class PyPylint(PythonPackage):
...
     depends_on('py-isort@4.2.5:', type=('build', 'run'))
     depends_on('py-isort@4.2.5:4.999', when='@2.3.1:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylint/package.py]: It's now `4.2.5:5.999`

File: var/spack/repos/builtin/packages/py-pylint/package.py [hunk 5]
```diff
@@ -39,3 +43,4 @@ class PyPylint(PythonPackage):
     depends_on('py-configparser', when='^python@:2.8', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylint/package.py:44]: Need setuptools-scm build dep

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-astroid/package.py (hunk 1 above)
Comment (scheibelp) [reply var/spack/repos/builtin/packages/py-astroid/package.py:16]: Hi @schmitts do you think you'll have a chance to address the comments here in the next few days?

Comment (schmitts) [reply var/spack/repos/builtin/packages/py-astroid/package.py:16]: Yes, will do, @scheibelp.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-astroid/package.py:16]: These dependencies still need to be updated:
```python
install_requires =
    lazy_object_proxy>=1.4.0
    wrapt>=1.11,<1.13
    typed-ast>=1.4.0,<1.5;implementation_name=="cpython" and python_version<"3.8"
python_requires = ~=3.6
```

Comment (schmitts) [reply var/spack/repos/builtin/packages/py-astroid/package.py:16]: Done

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**